src/CPULSSolver.cpp

```diff
@@ -0,0 +1,898 @@
+        double fission_source_y =
+            pairwise_sum<FP_PRECISION>(fission_sources_y, _num_groups);
+
+        FP_PRECISION* fission_sources_z = _groupwise_scratch.at(tid);
+        for (int g_prime=0; g_prime < _num_groups; g_prime++)
+          fission_sources_z[g_prime] =
+              nu_sigma_f[g_prime] * _scalar_flux_xyz(r,g_prime,2);
+        double fission_source_z =
+            pairwise_sum<FP_PRECISION>(fission_sources_z, _num_groups);
+
+        fission_source_x /= _k_eff;
+        fission_source_y /= _k_eff;
+        fission_source_z /= _k_eff;
+
+        src_x = scatter_source_x + chi[g] * fission_source_x;
+        src_y = scatter_source_y + chi[g] * fission_source_y;
+        src_z = scatter_source_z + chi[g] * fission_source_z;
+
+        /* Compute total (scatter+fission+fixed) reduced source moments */
+        if (_solve_3D) {
+          _reduced_sources_xyz(r,g,0) = ONE_OVER_FOUR_PI / 2 *
+              (_FSR_lin_exp_matrix[r*num_coeffs  ] * src_x +
+               _FSR_lin_exp_matrix[r*num_coeffs+2] * src_y +
+               _FSR_lin_exp_matrix[r*num_coeffs+3] * src_z);
+          _reduced_sources_xyz(r,g,1) = ONE_OVER_FOUR_PI / 2 *
+              (_FSR_lin_exp_matrix[r*num_coeffs+2] * src_x +
+               _FSR_lin_exp_matrix[r*num_coeffs+1] * src_y +
+               _FSR_lin_exp_matrix[r*num_coeffs+4] * src_z);
+          _reduced_sources_xyz(r,g,2) = ONE_OVER_FOUR_PI / 2 *
+              (_FSR_lin_exp_matrix[r*num_coeffs+3] * src_x +
+               _FSR_lin_exp_matrix[r*num_coeffs+4] * src_y +
+               _FSR_lin_exp_matrix[r*num_coeffs+5] * src_z);
+        }
+        else {
+          _reduced_sources_xyz(r,g,0) = ONE_OVER_FOUR_PI / 2 *
+              (_FSR_lin_exp_matrix[r*num_coeffs  ] * src_x +
+               _FSR_lin_exp_matrix[r*num_coeffs+2] * src_y);
+          _reduced_sources_xyz(r,g,1) = ONE_OVER_FOUR_PI / 2 *
+              (_FSR_lin_exp_matrix[r*num_coeffs+2] * src_x +
+               _FSR_lin_exp_matrix[r*num_coeffs+1] * src_y);
+        }
+      }
+    }
+  }
+}
+
+
+/**
+ * @brief Computes the contribution to the FSR scalar flux from a Track segment.
+ * @details This method integrates the angular flux for a Track segment across
+ *          energy groups and polar angles, and tallies it into the FSR
+ *          scalar flux, and updates the Track's angular flux.
+ * @param curr_segment a pointer to the Track segment of interest
+ * @param azim_index a pointer to the azimuthal angle index for this segment
+ * @param track_flux a pointer to the Track's angular flux
+ * @param fsr_flux a pointer to the temporary FSR flux buffer
+ * @param x the x-coord of the segment starting point
+ * @param y the y-coord of the segment starting point
+ * @param z the z-coord of the segment starting point
+ * @param fwd int indicating whether the segment is pointing forward (1) or
+ *            backwards (-1)
+//FIXME: NOT THE CORRECT PARAMS
```

Comment:
@param polar_index polar angle index for this segment
@param track_flux a pointer to the Track's angular flux
@param fsr_flux a pointer to the temporary FSR flux buffer
@param scratch_pad a pointer to a temporary working array
@param direction a vector with the track direction